crowbar_framework/app/models/service_object.rb

```diff
@@ -822,8 +805,17 @@ def apply_role(role, inst, in_queue)
     # We also check that all nodes we'll require are in the ready state.
     #
 
-    # Initialize variables used in ensure at the end of the method
-    chef_daemon_nodes = []
+    nodes_without_admin = []
+    NodeObject.find_all_nodes.each do |node|
```

Comment:
how much simpler would it be?

```ruby
NodeObject.find_all_nodes.select { node[:platform] != "windows" || !node.admin? }.map(&:name)
```
the `!admin?` part does not work somehow. so if you have a simpler version please tell me how and i will change it.